compositor_pipeline/src/pipeline.rs

```diff
@@ -31,6 +33,19 @@ pub mod input;
 pub mod output;
 pub mod structs;
 
+pub struct ExactPort(pub u16);
+
+pub enum Port {
```

Comment:
If you want to distinguish those types maybe change `Port` to `PortRequest`/`RequestedPort`/`PortRange`